src/peer_select.cc

```diff
@@ -1152,9 +1157,7 @@ PeerSelector::interestedInitiator()
 
 bool
 PeerSelector::wantsMoreDestinations() const {
-    const auto maxCount = Config.forward_max_tries;
-    return maxCount >= 0 && foundPaths <
-           static_cast<std::make_unsigned<decltype(maxCount)>::type>(maxCount);
+    return foundPaths < static_cast<size_t>(Config.forward_max_tries);
```

Comment:
let's check. My concern here is that size_t may be signed on some platforms. I'll accept the suggestion and iterate if the build fails there

Reply:
I doubt we still support platforms with signed size_t (for other reasons). To avoid conversion and overflow bugs (throughout the code), I would consider statically asserting that size_t is unsigned (as [required](https://stackoverflow.com/questions/1089176/is-size-t-always-unsigned) by modern C++). I do not insist on that static assertion addition to this PR.

Reply:
I agree in principle, but in practice I recall there was a case which we're supposed to support for another 3 months or so. Let's see what happens; worst case I will just iterate

Reply:
Glad we agree in principle!

> but in practice I recall there was a case which we're supposed to support for another 3 months or so

In my experience, such vague/citation-free recollections are often misleading -- humans often misunderstand/mischaracterize a difficult problem, but their fix "works", confirming (in their mind) the incorrect interpretation of the situation. You may be right in this particular case, of course, but given the lack of citations, I have to go with whatever I can find myself. I see several [references](https://stackoverflow.com/a/206584) to GCC prior to v2.4 using signed size_t (in some cases) and switching to unsigned size_t from v2.4. We cannot compile Squid with GCC v2, so that particular danger should not apply. I could not find any references to similar problems in clang (which does not surprise me since clang is a "newer" project).

Reply:
> In my experience, such vague/citation-free recollections are often misleading -- humans often misunderstand/mischaracterize a difficult problem, but their fix "works", confirming (in their mind) the incorrect interpretation of the situation. You may be right in this particular case, of course, but given the lack of citations, I have to go with whatever I can find myself. I see several [references](https://stackoverflow.com/a/206584) to GCC prior to v2.4 using signed size_t (in some cases) and switching to unsigned size_t from v2.4. We cannot compile Squid with GCC v2, so that particular danger should not apply. I could not find any references to similar problems in clang (which does not surprise me since clang is a "newer" project).

I also do not trust such vague recollections, so I was happy to try and iterate in case of failure. I is quite possible (likely, even) that the problem I recall was with a different piece of code. Anyway, it seems this particular case is working fine; now checking the rest :)